obspy/signal/tests/test_spectral_estimation.py

```diff
@@ -677,6 +677,32 @@ def test_ppsd_temporal_plot(self):
             ppsd.plot_temporal([0.1, 1, 10], filename=ic.name, show=False,
                                **restrictions)
 
+    def test_exclude_last_sample(self):
+
+        header = {
+            "starttime":  UTCDateTime("2017-01-01T00:00:00"),
+            "network": "GR",
+            "station": "FUR",
+            "channel": "BHZ"
+        }
+
+        # 49 segments of 30 minutes to allow 30 minutes overlap in next day
+        tr = Trace(data=np.arange(30 * 60 * 4, dtype=np.int32), header=header)
+
+        ppsd = PPSD(
+            tr.stats,
+            read_inventory()
+        )
+
+        ppsd.add(tr)
+
+        self.assertEqual(3, len(ppsd._times_processed))
+        self.assertEqual(3600, ppsd.len)
+
+        for i, time in enumerate(ppsd._times_processed):
+            current = UTCDateTime("2017-01-01T00:00:00") + (i * 30 * 60)
+            self.assertTrue(UTCDateTime(time) == current)
```

Comment:
interesting.. so I guess start times of individual PSD slices were never tested so far.. :-/